packages/elements/src/pagination/index.ts

```diff
@@ -367,21 +370,15 @@ export class Pagination extends BasicElement {
   /**
    * @override
    */
-  protected update (changedProperties: PropertyValues): void {
-    super.update(changedProperties);
-
-    if (changedProperties.has('value')) {
-      this.input.setAttribute('aria-valuenow', this.internalValue.toString());
-    }
-
-    if (changedProperties.has('max')) {
-      if (!this.infinitePaginate) {
-        this.input.setAttribute('aria-valuemax', this.max);
+  protected firstUpdated (changedProperties: PropertyValues): void {
+    super.firstUpdated(changedProperties);
+    // Prevent copy/paste no to the text input
+    this.input.addEventListener('paste', (event: ClipboardEvent) => {
```

Comment:
This code probably does not cover text drag&drop?

But anyway, you already do two types of input validation: `onKeyDown` and `paste`. Why not to simply stripe invalid characters as follows:

```
@input=${this.onInputInput}

protected stripeInvalidCharacters (value: string): string {
  return value.replace(/[^\d]/g, '');
}

/**
   * Runs on input element `input` event
   * @param event `input` event
   * @returns {void}
   */
  protected onInputInput (event: InputEvent): void {
    const currentInput = this.inputValue;
    const inputValue = this.stripeInvalidCharacters;
    if (currentInput !== inputValue) {
      this.inputValue = inputValue;
    }
  }

```

Reply:
I will try, it could be better if group it in one place.

Reply:
Implemented, I add the condition to handle the invalid number starting with `0` as well